react/features/calendar-sync/functions.any.js

```diff
@@ -0,0 +1,173 @@
+                // occurrences of a recurring event or repetitions of an even
+                // from multiple calendars.
+                new Date(entry.startDate).toTimeString()
+            ]));
+            const existingEntry = entryMap.get(key);
+
+            // We want only the earliest occurrence (which hasn't ended in the
+            // past, that is) of a recurring event.
+            if (!existingEntry || existingEntry.startDate > entry.startDate) {
+                entryMap.set(key, entry);
+            }
+        }
+    }
+
+    dispatch(
+        setCalendarEvents(
+            Array.from(entryMap.values())
+                .sort((a, b) => a.startDate - b.startDate)
+                .slice(0, MAX_LIST_LENGTH)));
+}
+
+/**
+ * Updates the calendar entries in Redux when new list is received.
+ *
+ * @param {Object} event - An event returned from the native calendar.
+ * @param {Array<string>} knownDomains - The known domain list.
+ * @private
+ * @returns {CalendarEntry}
+ */
+function _parseCalendarEntry(event, knownDomains) {
+    if (event) {
+        const url = _getURLFromEvent(event, knownDomains);
+
+        if (url) {
+            const startDate = Date.parse(event.startDate);
+            const endDate = Date.parse(event.endDate);
+
+            if (isNaN(startDate) || isNaN(endDate)) {
+                logger.warn(
+                    'Skipping invalid calendar event',
+                    event.title,
+                    event.startDate,
+                    event.endDate
+                );
+            } else {
+                return {
+                    endDate,
+                    id: event.id,
+                    startDate,
+                    title: event.title,
+                    url
+                };
+            }
+        }
+    }
+
+    return null;
+}
+
+/**
+ * Retrieves a Jitsi Meet URL from an event if present.
+ *
+ * @param {Object} event - The event to parse.
+ * @param {Array<string>} knownDomains - The known domain names.
+ * @private
+ * @returns {string}
+ */
+function _getURLFromEvent(event, knownDomains) {
+    const linkTerminatorPattern = '[^\\s<>$]';
+    const urlRegExp
+        = new RegExp(
+        `http(s)?://(${knownDomains.join('|')})/${linkTerminatorPattern}+`,
+        'gi');
+    const schemeRegExp
+        = new RegExp(`${APP_LINK_SCHEME}${linkTerminatorPattern}+`, 'gi');
+    const fieldsToSearch = [
+        event.title,
+        event.url,
+        event.location,
+        event.notes,
+        event.description
+    ];
+
+    for (const field of fieldsToSearch) {
+        if (typeof field === 'string') {
+            const matches = urlRegExp.exec(field) || schemeRegExp.exec(field);
+
+            if (matches) {
+                const url = parseURIString(matches[0]);
+
+                if (url) {
+                    return url.toString();
+                }
+            }
+        }
+    }
+
+    return null;
+}
```

Comment:
Yes, there was only one change I made. https://github.com/jitsi/jitsi-meet/pull/3340/commits/7f8a1e2c1799e6bb78a3445e2c3db2f88f6d91fe#diff-e626ccf6b0437daab4a769da852e453dR24
It was executing one unnecessary fetch every time because of comparing arrays, at least that what I saw on mobile, not sure whether that array === array is working differently on mobile.

Reply:
I see. So instead of comparing array values it was comparing if they're pointers to the same array.